ui/src/common/withRouteDataFetcher.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { getWrapperComponentDisplayName } from './utils';
+import LoadingOrChildren from './components/LoadingOrChildren';
+
+export default function withRouteDataFetcher(
```

Comment:
the name is a bit confusing tbh, wouldn't be more obvious something like ``withLocationParamsDataFetcher`` or similar?

Reply:
this is used to `fetch` for `data` dependencies of current `route`. So name made sense 😄

Reply:
To be honest I am more concern about the fact that some components are dispatching actions which doesn't even use params, so it looks little awkward.